engine/src/main/java/org/terasology/rendering/nui/contextMenu/ContextMenuBuilder.java

```diff
@@ -20,20 +20,18 @@
 import org.terasology.math.geom.Vector2i;
 import org.terasology.rendering.nui.NUIManager;
 import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.widgets.UIList;
 import org.terasology.rendering.nui.widgets.UpdateListener;
 
 import java.util.List;
 
-/**
- * A builder class to construct and show {@link ContextMenuScreen} instances.
- * <p>
- * Should be used in favor of manually creating the screen.
- */
 public class ContextMenuBuilder {
```

Comment:
A builder with a single setter and a getResult method looks strange to me. I would go for a utility class with a single public static method `void showContextMenu(NUIManager manager, Vector2i position, ContextMenuTree tree)`. It would then still have some private methods that it uses for the construction of the context menu.


It isn't much like a builder anymore. (See https://en.wikipedia.org/wiki/Builder_pattern)